File: sdkwork-ai-sdk-go/types/chat_completion_request.go
package types

// Chat completion request
type ChatCompletionRequest struct {
	Model string `json:"model"`
	Messages []ChatMessage `json:"messages"`
	Temperature float64 `json:"temperature"`
	N int `json:"n"`
	Stream bool `json:"stream"`
	Stop []string `json:"stop"`
	User string `json:"user"`
	Tools []Tool `json:"tools"`
	ToolChoice interface{} `json:"toolChoice"`
	Seed float64 `json:"seed"`
	StreamOptions StreamOptions `json:"streamOptions"`
	Modalities []string `json:"modalities"`
	Audio AudioParameters `json:"audio"`
	Store bool `json:"store"`
	Metadata map[string]string `json:"metadata"`
	Thinking Thinking `json:"thinking"`
	TopP float64 `json:"top_p"`
	MaxTokens int `json:"max_tokens"`
	MaxCompletionTokens int `json:"max_completion_tokens"`
	PresencePenalty float64 `json:"presence_penalty"`
	FrequencyPenalty float64 `json:"frequency_penalty"`
	LogitBias map[string]int `json:"logit_bias"`
	ResponseFormat ResponseFormat `json:"response_format"`
	ServiceTier string `json:"service_tier"`
	ReasoningEffort string `json:"reasoning_effort"`
	ParallelToolCalls bool `json:"parallel_tool_calls"`
	Logprobs bool `json:"logprobs"`
	TopLogprobs int `json:"top_logprobs"`
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/models/chat_completion_request.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class ChatCompletionRequest:
    """Chat completion request"""
    model: str
    messages: List[ChatMessage]
    temperature: float = None
    n: int = None
    stream: bool = None
    stop: List[str] = None
    user: str = None
    tools: List[Tool] = None
    tool_choice: Any = None
    seed: float = None
    stream_options: StreamOptions = None
    modalities: List[str] = None
    audio: AudioParameters = None
    store: bool = None
    metadata: Dict[str, str] = None
    thinking: Thinking = None
    top_p: float = None
    max_tokens: int = None
    max_completion_tokens: int = None
    presence_penalty: float = None
    frequency_penalty: float = None
    logit_bias: Dict[str, int] = None
    response_format: ResponseFormat = None
    service_tier: str = None
    reasoning_effort: str = None
    parallel_tool_calls: bool = None
    logprobs: bool = None
    top_logprobs: int = None
